girder-dandi-archive/girder_dandi_archive/rest.py

```diff
@@ -0,0 +1,64 @@
+from girder.api import access
+from girder.api.rest import Resource
+from girder.api.describe import describeRoute, Description
+from girder.models.setting import Setting
+from girder.models.folder import Folder
+from girder.exceptions import RestException
+
+from .util import (
+    DANDISET_ID_COUNTER,
+    staging_collection,
+    pad_dandiset_id,
+    validate_dandiset_id,
+)
+
+
+class DandiResource(Resource):
+    def __init__(self):
+        super(DandiResource, self).__init__()
+
+        self.resourceName = "dandi"
+        self.route("GET", (), self.get_dandiset)
+        self.route("POST", (), self.create_dandiset)
+
+    @access.user
+    @describeRoute(
+        Description("Create Dandiset")
+        .param("name", "Name of the Dandiset.")
+        .param("description", "Description of the Dandiset.")
+    )
+    def create_dandiset(self, params):
+        name, description = params["name"], params["description"]
+        exists = Folder().findOne({"name": name})
+
+        if exists:
+            raise RestException("Dandiset already exists", code=409)
+
+        current = Setting().get(DANDISET_ID_COUNTER)
+        if current is None:
+            new_id_count = Setting().set(DANDISET_ID_COUNTER, 0)
+        else:
+            new_id_count = Setting().set(DANDISET_ID_COUNTER, current + 1)
+
+        padded_id = pad_dandiset_id(new_id_count["value"])
+        meta = {"name": name, "description": description, "id": padded_id}
+
+        staging = staging_collection()
+        folder = Folder().createFolder(
+            staging,
+            name,
+            parentType="collection",
+            creator=self.getCurrentUser(),
+            public=False,
+        )
+        folder = Folder().setMetadata(folder, {"dandiset": meta})
```

Comment:
Why are you duplicating the `name` (and potentially `description`) into the metadata, when it already exists on the top-level folder itself?

Reply:
The `name` and `description` fields are just the initial required parts of a larger schema that's stored in this metadata. Because the idea is to have all metadata of a dandiset live under `meta.dandiset`, I don't think it makes sense to separate this.